src/app/script/app_clipboard_object.cpp

```diff
@@ -0,0 +1,138 @@
+// Aseprite
+// Copyright (c) 2022-2024  Igara Studio S.A.
+//
+// This program is distributed under the terms of
+// the End-User License Agreement for Aseprite.
+
+#ifdef HAVE_CONFIG_H
+  #include "config.h"
+#endif
+
+#include "app/modules/palettes.h"
+#include "app/script/luacpp.h"
+#include "app/util/clipboard.h"
+#include "engine.h"
+#include "clip/clip.h"
+
+namespace app {
+namespace script {
+
+namespace {
+
+struct Clipboard {};
+
+int Clipboard_clear(lua_State* L)
+{
+  if (!clip::clear())
+    return luaL_error(L, "failed to clear the clipboard");
+
+  return 1;
+}
+
+int Clipboard_is_text(lua_State* L)
+{
+  lua_pushboolean(L, clip::has(clip::text_format()));
+  return 1;
+}
+
+int Clipboard_is_image(lua_State* L)
+{
+  lua_pushboolean(L, clip::has(clip::image_format()));
+  return 1;
+}
+
+int Clipboard_is_empty(lua_State* L)
+{
+  // Using clip::has(clip::empty_format()) had inconsistent results, might as well avoid false positives
+  // by just checking the two formats we support
+  lua_pushboolean(
+    L, !clip::has(clip::image_format()) && !clip::has(clip::text_format())
+  );
+  return 1;
+}
+
+int Clipboard_get_image(lua_State* L)
+{
+  doc::Image* image = nullptr;
+  doc::Mask* mask = nullptr;
+  doc::Palette* palette = nullptr;
+  doc::Tileset* tileset = nullptr;
+  const bool result = app::Clipboard::instance()->getNativeBitmap(
+    &image, &mask, &palette, &tileset
+  );
+
+  // TODO: If we get a tileset, should we convert it to an image?
+  if (image == nullptr || !result)
+    return luaL_error(L, "failed to get image from clipboard");
+
+  push_image(L, image);
+  return 1;
+}
+
+int Clipboard_get_text(lua_State* L)
+{
+  std::string str;
+  if (!clip::get_text(str))
+    return luaL_error(L, "failed to get text from clipboard");
+
+  lua_pushstring(L, str.c_str());
+  return 1;
+}
+
+int Clipboard_set_image(lua_State* L)
+{
+  auto* image = may_get_image_from_arg(L, 2);
+  if (!image)
+    return luaL_error(L, "invalid image");
+
+  const bool result = app::Clipboard::instance()->setNativeBitmap(
+    image,
+    nullptr,
+    get_current_palette(),  // TODO: Not sure if there's any way to get the palette from the image
+    nullptr,
```

Comment:
This is the best we got at the moment. If we specify an isolated image.

Anyway it'd be probably better to use the active context site (the app::Site contains the active palette and tileset).